code/datums/quirks/neutral_quirks/transhumanist.dm

```diff
@@ -0,0 +1,147 @@
+#define MOOD_CATEGORY_TRANSHUMANIST_PEOPLE "transhumanist_people"
+#define MOOD_CATEGORY_TRANSHUMANIST_BODYPART "transhumanist_bodypart"
+// The number of silicons minus the number of organics determines the level
+#define TRANSHUMANIST_LEVEL_ECSTATIC 5
+#define TRANSHUMANIST_LEVEL_HAPPY 2
+#define TRANSHUMANIST_LEVEL_NEUTRAL 0
+#define TRANSHUMANIST_LEVEL_UNHAPPY -2
+#define TRANSHUMANIST_LEVEL_ANGRY -5
+
+
+/datum/quirk/transhumanist
+	name = "Transhumanist"
+	desc = "You see silicon life as the perfect lifeform and despise organic flesh. You are happier around silicons, but get frustrated when around organics. You seek to replace your fleshy limbs with their silicon counterparts. You start with a robotic limb."
+	icon = FA_ICON_ROBOT
+	quirk_flags = QUIRK_HUMAN_ONLY|QUIRK_PROCESSES|QUIRK_MOODLET_BASED
+	value = 0
+	gain_text = span_notice("You have a desire to ditch your feeble organic flesh and surround yourself with robots.")
+	lose_text = span_danger("Robots don't seem all that great anymore.")
+	medical_record_text = "Patient reports hating pathetic creatures of meat and bone."
+	mail_goodies = list(
+		/obj/item/stock_parts/cell/potato,
+		/obj/item/stack/cable_coil,
+		/obj/item/toy/talking/ai,
+		/obj/item/toy/figure/borg,
+	)
+	var/slot_string
+	var/obj/item/bodypart/old_limb
+
+/datum/quirk/transhumanist/add(client/client_source)
+	RegisterSignal(quirk_holder, COMSIG_CARBON_POST_ATTACH_LIMB, PROC_REF(calculate_bodypart_score))
+	RegisterSignal(quirk_holder, COMSIG_CARBON_POST_REMOVE_LIMB, PROC_REF(calculate_bodypart_score))
+	RegisterSignal(quirk_holder, COMSIG_CARBON_GAIN_ORGAN, PROC_REF(calculate_bodypart_score))
+	RegisterSignal(quirk_holder, COMSIG_CARBON_LOSE_ORGAN, PROC_REF(calculate_bodypart_score))
+	calculate_bodypart_score()
+
+/datum/quirk/transhumanist/remove()
+	UnregisterSignal(quirk_holder, list(COMSIG_CARBON_REMOVE_LIMB, COMSIG_CARBON_ATTACH_LIMB))
+
+/datum/quirk/transhumanist/proc/calculate_bodypart_score()
+	SIGNAL_HANDLER
+	var/mob/living/carbon/human/human_holder = quirk_holder
+	var/organic_bodytypes = 0
+	var/silicon_bodytypes = 0
+	var/other_bodytypes = FALSE
+	for(var/obj/item/bodypart/part as anything in human_holder.bodyparts)
+		if(part.bodytype & BODYTYPE_ROBOTIC)
+			silicon_bodytypes += 1
+		else if(part.bodytype & BODYTYPE_ORGANIC)
+			organic_bodytypes += 0.1
+		else
+			other_bodytypes = TRUE
+
+	for(var/obj/item/organ/organ as anything in human_holder.organs)
+		if(organ.organ_flags & ORGAN_ROBOTIC)
+			silicon_bodytypes += 0.25
+		else if(organ.organ_flags & ORGAN_ORGANIC)
+			organic_bodytypes += 0.02
+
+	if(!other_bodytypes)
+		if(organic_bodytypes == 0)
+			quirk_holder.add_mood_event(MOOD_CATEGORY_TRANSHUMANIST_BODYPART, /datum/mood_event/completely_robotic)
```

Comment:
Mhm... you will never be completely robotic, unless you find a way to make your brain robotic, which is only possible by species-swapping into an android. This is but a comment anyway.